### ISSUE TITLE
[key-spacing] key-spacing of interfaces and other TS-specific key-value pairs

### ISSUE THREAD

**Issue by lolmaus** [typescript-eslint/typescript-eslint#940]
**Repro**

```JSON
{
  "rules": {
    "key-spacing": ["error", { "align": "value" }]
  }
}
```

```TS
interface Foo {
  foo: string;
  barBazQuux: string;
}
```

**Expected Result**

Should be highlighted as an error and auto fixed into:

```TS
interface Foo {
  foo:        string;
  barBazQuux: string;
}
```

**Actual Result**

ESLint ignores interfaces.

**Additional Info**

Time for `@typescript-eslint/key-spacing`?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.6.2` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.9.0` |
| `yarn`                              | `1.17.3` |

**Comment by bradzacher** [comment]
Is that the desired result that we want - the `key-spacing` rule specifically states "object literal properties".

In vanilla JS, do you do the same thing with classes?

In TS would you want the same functionality with enums? object type literals?

**Comment by lolmaus** [comment]
I want to be able to auto-align every key-value pair, similar to what ESLint's `key-spacing` does.

Properly aligned code is easier to navigate with eyes.

**Comment by kamil-augustyniak** [comment]
`key-spacing` for enums would be great option too.

**Comment by AnthonyLzq** [comment]
Is there any update about this?

**Comment by bradzacher** [comment]
With any issue in opened in this project - it either has a visible progress in the form of an attached PR, or it has no progress.

We are a community run project. The **volunteer** maintainers spend most of their time triaging issues and reviewing PRs. This means that most issues *will not progress* unless a member of the community steps up and **champions** it.

If this issue is important to you - consider being that champion.

If not - please just subscribe to the issue and wait patiently.
Commenting asking for status updates does not bump issue priority in any way and just serves to spam ***everyone*** subscribed to the issue.

**Comment by dvakatsiienko** [comment]
+1

**Comment by B2D1** [comment]
Looks like this issue has'not been closed even in 2022!

**Comment by AlbinoGeek** [comment]
So, I use the vs code extension called alignment, and it gets you about
halfway to this feature, but you have to disable format on save, or the
winter will undo it.....

**Comment by drichard-nexapp** [comment]
For enum there is @typescript-eslint/space-infix-ops that add spacing in enums

**Comment by kastamas** [comment]
any updates?

**Comment by gsouf** [comment]
@JamesHenry  I want to fix this issue. Can you please give me any direction to find the right place in the codebase to do this?
